Alpha setting for `SerialisableDrawable`

NiyazBiyaz opened the pull request:
It was allowed in #22475, but not implemented yet.

---
## Preview
https://github.com/user-attachments/assets/460dce8e-9a0a-4243-800c-3e7c2feabe62

- [x] Can adjust alpha
- [x] Can save it
- [x]  Works for all skin HUD components

## Issues

### Composite drawables
> Of note though, some components will not look good with alpha applied (unless buffered) due to alpha being applied individually to each layer.

Some components looks bad with low alpha value. At this moment fixed only `BarHitErrorMeter` just using `BufferedContainer` for some static inner layers, but i have no idea how to fix other, because it's create problems with colors and updating.

##### Argon score counters
<img width="1087" height="308" alt="All Argon Counters" src="https://github.com/user-attachments/assets/edde3af9-492a-4a08-8003-c7f815a9ac0f" />

##### Some other composite drawables
<img width="508" height="525" alt="AimHitErrorMeter" src="https://github.com/user-attachments/assets/a155cb1d-a7db-43d8-a2d7-d3b63af49fb8" />

If acceptable it can be be left as is.

### `SongProgress` animations
They related with the `Alpha` changing, and adjusting value of this doesn't work after reload. (Also alpha setting value on sidebar have broken initialization with `SongProgress` objects) 

[SongProgress.cs](https://github.com/ppy/osu/blob/master/osu.Game/Screens/Play/HUD/SongProgress.cs#L95)
```csharp
        protected override void PopIn() => this.FadeIn(500, Easing.OutQuint);

        protected override void PopOut() => this.FadeOut(100);
```
I'm not sure i can just delete them. So i want to know what to do about it, because it's objective problem.

----

_First PR. I read contributing guidelines but can't run `CodeInspect.sh` (i cannot install dotnet 6.0 in my OS for `CodeFileSanity`, and in some reason it's all can't get assembly references)... but i hope i didn't break anything really important, because tests in visual browser and game itself works fine._

NiyazBiyaz committed:
Add alpha value changing for skin HUD components

- Field serialization/deserialization

- Default sliderbar for every HUD component

NiyazBiyaz committed:
Merge branch 'ppy:master' into alpha-setting-for-serialisable-drawable

NiyazBiyaz committed:
Fix `BarHitErrorMeter` color bars mixing

NiyazBiyaz committed:
Merge branch 'ppy:master' into alpha-setting-for-serialisable-drawable

peppy commented:
Alpha doesn't work great for some components due to missing frame buffer, touched on in the OP but without performance overhead considerations.

> They related with the Alpha changing, and adjusting value of this doesn't work after reload. (Also alpha setting value on sidebar have broken initialization with SongProgress objects)

Need to move alpha usage to not `this`, with a nested container or otherwise, assuming that's correctly supported by the game.

Dunno

NiyazBiyaz committed:
Redirect FadeIn/Out animations for each Child

NiyazBiyaz committed:
Add condition for disable Alpha adjusting in some objects

NiyazBiyaz committed:
Disable Alpha adjusting for some objects

- ArgonAccuracyCounter
- ArgonComboCounter
- ArgonPerformancePointsCounter
- ArgonScoreCounter
- ArgonUnstableRateCounter
- DefaultRankDisplay
- LegacySongProgress
- AimErrorMeter

May be needed to expand this list considering new components in the upstream branch as next PR
